### PR TITLE
feat: implement authn

### PR DESCRIPTION
**Pull request by timonmasberg**
- implements oauth in our spa
- implements header authentication in our api based on the azure oauth proxy headers
- adds spectator as angular testing helper
- adapts ci pipelines to deploy authentication with preview deployments

testuser: 
username: testuser
password: testuser1234

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -32400,62 +32160,6 @@
...
-      "dev": true,
-      "optional": true
-    },
-    "@nrwl/nx-win32-x64-msvc": {
-      "version": "15.8.6",
-      "resolved": "https://registry.npmjs.org/@nrwl/nx-win32-x64-msvc/-/nx-win32-x64-msvc-15.8.6.tgz",
-      "integrity": "sha512-P0Sb4HJCeoeTvPFdUMKljRUIjzso0go36cn1Zpl+Z5CG/nbOvLlbnzh6rg15SRNu9OLWTHNPtyQIvKxjqEDoxg==",
-      "dev": true,
-      "optional": true
-    },
```
Comment (Cammisuli) [inline package-lock.json:32458]: This is interesting. These shouldn't have been removed. Thats why they don't get installed on the CI

Comment (timonmasberg) [reply package-lock.json:32458]: let me try to remove the lockfile and node_modules and regenerate it, maybe something went wrong on my end

Comment (timonmasberg) [reply package-lock.json]: https://github.com/kordis-leitstelle/kordis/actions/runs/4439289647/jobs/7791543255 well... idk why this is so flocky, but it seems something went wrong when we migrated or something. Idk if that might be related to the original issue. Hope the insights you got here helped! @Cammisuli

**Comment by Cammisuli** [comment]
yay it worked 🙂

**Review by JSPRH — DISMISSED** [review]

**Review by JSPRH — APPROVED** [review]

**Pull request merged**